agent/agent.go

```diff
@@ -8,30 +8,33 @@ import (
 	"sync"
 	"time"
 
+	"github.com/hashicorp/consul-template/manager"
 	consul "github.com/hashicorp/consul/api"
 	"github.com/pkg/errors"
-	"github.com/hashicorp/consul-template/manager"
+	"github.com/trento-project/trento/agent/discover"
 )
 
 type Agent struct {
 	cfg              Config
 	check            Checker
+	discovers        []discover.Discovery
 	consulResultChan chan CheckResult
 	wsResultChan     chan CheckResult
 	webService       *webService
-	consul           *consul.Client
+	consul           consul.Client
 	ctx              context.Context
 	ctxCancel        context.CancelFunc
 	templateRunner   *manager.Runner
 }
 
 type Config struct {
+	CheckerTTL          time.Duration
 	WebHost             string
 	WebPort             int
 	ServiceName         string
 	InstanceName        string
 	DefinitionsPath     string
-	TTL                 time.Duration
+	DiscoverInterval    time.Duration
```

Comment:
I guess that this eventually will become a list of `time.Duration`. If we have multiple discoveries, not all of them might need to have the same interval.
Just an observation. We can change this in the future

Reply:
Thats a possibility.